prime-router/src/test/kotlin/fhirengine/engine/fhirRouterTests/BundleUpdateTests.kt

```diff
@@ -0,0 +1,156 @@
+import io.mockk.clearAllMocks
+import io.mockk.mockkClass
+import io.mockk.spyk
+import org.hl7.fhir.r4.model.Endpoint
+import org.hl7.fhir.r4.model.Provenance
+import org.jooq.tools.jdbc.MockConnection
+import org.jooq.tools.jdbc.MockDataProvider
+import org.jooq.tools.jdbc.MockResult
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.TestInstance
+import java.util.UUID
+import kotlin.test.Test
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class BundleUpdateTests {
+    val dataProvider = MockDataProvider { emptyArray<MockResult>() }
+    val connection = MockConnection(dataProvider)
+    val accessSpy = spyk(DatabaseAccess(connection))
+    val blobMock = mockkClass(BlobAccess::class)
+    val queueMock = mockkClass(QueueAccess::class)
+    val metadata = Metadata(schema = Schema(name = "None", topic = "full-elr", elements = emptyList()))
+    val bodyUrl = "http://anyblob.com"
+    private val defaultReceivers = listOf(
+        Receiver(
+            "full-elr-hl7",
+            "co-phd",
+            "full-elr",
+            CustomerStatus.ACTIVE,
+            "one"
+        ),
+        Receiver(
+            "full-elr-hl7-2",
+            "co-phd",
+            "full-elr",
+            CustomerStatus.INACTIVE,
+            "one"
+        )
+    )
+    val oneOrganization = DeepOrganization(
+        "co-phd",
+        "test",
+        Organization.Jurisdiction.FEDERAL,
+        receivers = defaultReceivers
+    )
+
+    val message = spyk(RawSubmission(UUID.randomUUID(), "http://blob.url", "test", "test-sender"))
+
+    private val validFhirWithProvenance = """
+    {
+        "resourceType": "Bundle",
+        "id": "1666038428133786000.94addcb6-835c-4883-a095-0c50cf113744",
+        "meta": {
+        "lastUpdated": "2022-10-17T20:27:08.149+00:00",
+        "security": [
+            {
+                "code": "SECURITY",
+                "display": "SECURITY"
+            }
+        ]
+    },
+        "identifier": {
+        "value": "MT_COCAA_ORU_AAPHELR.1.6214638"
+    },
+        "type": "message",
+        "timestamp": "2028-08-08T15:28:05.000+00:00",
+        "entry": [
+            {
+                "fullUrl": "Provenance/1666038430962443000.9671377b-8f2b-4f5c-951c-b43ca8fd1a25",
+                "resource": {
+                    "resourceType": "Provenance",
+                    "id": "1666038430962443000.9671377b-8f2b-4f5c-951c-b43ca8fd1a25",
+                    "occurredDateTime": "2028-08-08T09:28:05-06:00",
+                    "recorded": "2028-08-08T09:28:05-06:00",
+                    "activity": {
+                        "coding": [
+                            {
+                                "system": "http://terminology.hl7.org/CodeSystem/v2-0003",
+                                "code": "R01",
+                                "display": "ORU_R01"
+                            }
+                        ]
+                    }
+                }
+            }
+        ]
+    }"""
```

Comment:
(Question) Do we manually generate FHIR and other input for tests, or do we pull those from somewhere?

Reply:
Whatever works best for you. I generated these by running an HL7 message through the universal pipeline, downloading the resulting FHIR, and picking the parts I wanted.